src/contracts/storage/pool.rs

```diff
@@ -186,7 +186,8 @@ impl Pool {
         total_amount_in: &mut TokenAmount,
         protocol_fee: Percentage,
         fee_tier: FeeTier,
-    ) {
+    ) -> (bool, Option<i32>) {
+        let mut has_crossed = (false, None);
```

Comment:
if case tick was crossed we know which one. `tick_index` is stored in `limiting_tick`